driver-sync/src/test/functional/com/mongodb/client/RetryableWritesProseTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client;
+
+import com.mongodb.MongoException;
+import org.bson.Document;
+import org.junit.Before;
+import org.junit.Test;
+
+import static com.mongodb.ClusterFixture.getServerStatus;
+import static com.mongodb.ClusterFixture.isDiscoverableReplicaSet;
+import static com.mongodb.ClusterFixture.isSharded;
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static com.mongodb.ClusterFixture.serverVersionLessThan;
+import static java.util.Arrays.asList;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+
+public class RetryableWritesProseTest extends DatabaseTestCase {
+
+    @Before
+    @Override
+    public void setUp() {
+        assumeTrue(canRunTests());
+        super.setUp();
+    }
+
+    @Test
+    public void testRetryWritesAgainstMMAPv1RaisesError() {
+        boolean exceptionFound = false;
+
+        try {
+            collection.insertOne(Document.parse("{x: 1}"));
+        } catch (MongoException e) {
```

Comment:
Catch MongoClientException and change assertion accordingly.  Might as well assert about both the code and the message of the exception that is the cause of the MongoClientException (use `Throwable#getCause` to access it).

Reply:
Done.